tests/flows/teleporter_registry.go

```diff
@@ -0,0 +1,221 @@
+	)
+
+	// Deploy the new version of Teleporter to both chains
+	newTeleporterAddress := deployNewTeleporterVersion(ctx, network, fundedKey)
+
+	// Create chain config file with off chain message for each chain
+	offchainMessageA, warpEnabledChainConfigA := initChainConfig(subnetAInfo, newTeleporterAddress)
+	offchainMessageB, warpEnabledChainConfigB := initChainConfig(subnetBInfo, newTeleporterAddress)
+
+	// Create chain config with off chain messages
+	chainConfigs := make(map[string]string)
+	setChainConfig(chainConfigs, subnetAInfo, warpEnabledChainConfigA)
+	setChainConfig(chainConfigs, subnetBInfo, warpEnabledChainConfigB)
+
+	// Restart nodes with new chain config
+	nodeNames := network.GetSubnetNodeNames()
+	network.RestartNodes(ctx, nodeNames, runner_sdk.WithChainConfigs(chainConfigs))
+
+	utils.AddProtocolVersionAndWaitForAcceptance(
+		ctx,
+		network,
+		subnetBInfo,
+		subnetAInfo,
+		newTeleporterAddress,
+		fundedKey,
+		offchainMessageB)
+
+	// Send a message using old Teleporter version to example messenger using new Teleporter version.
+	// Message should be received successfully since we haven't updated mininum Teleporter version yet.
+	utils.SendExampleCrossChainMessageAndVerify(
+		ctx,
+		network,
+		subnetAInfo,
+		exampleMessengerA,
+		subnetBInfo,
+		exampleMessengerContractB,
+		exampleMessengerB,
+		fundedKey,
+		true)
+
+	// Update minimum Teleporter version on destination chain
+	opts, err := bind.NewKeyedTransactorWithChainID(fundedKey, subnetBInfo.EVMChainID)
+	Expect(err).Should(BeNil())
+
+	latestVersion, err := subnetBInfo.TeleporterRegistry.LatestVersion(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+	minTeleporterVersion, err := exampleMessengerB.GetMinTeleporterVersion(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+	tx, err := exampleMessengerB.UpdateMinTeleporterVersion(opts, latestVersion)
+	Expect(err).Should(BeNil())
+
+	receipt := utils.WaitForTransactionSuccess(ctx, subnetBInfo, tx)
+	minTeleporterVersionUpdatedEvent, err := utils.GetEventFromLogs(
+		receipt.Logs,
+		exampleMessengerB.ParseMinTeleporterVersionUpdated)
+	Expect(err).Should(BeNil())
+	Expect(minTeleporterVersionUpdatedEvent.OldMinTeleporterVersion.Cmp(minTeleporterVersion)).Should(Equal(0))
+	Expect(minTeleporterVersionUpdatedEvent.NewMinTeleporterVersion.Cmp(latestVersion)).Should(Equal(0))
+
+	utils.SendExampleCrossChainMessageAndVerify(
+		ctx,
+		network,
+		subnetAInfo,
+		exampleMessengerA,
+		subnetBInfo,
+		exampleMessengerContractB,
+		exampleMessengerB,
+		fundedKey,
+		false)
+
+	// Update the subnets to use new Teleporter messengers
+	network.SetTeleporterContractAddress(newTeleporterAddress)
+	subnetAInfo, subnetBInfo = utils.GetTwoSubnets(network)
+	utils.SendExampleCrossChainMessageAndVerify(
+		ctx,
+		network,
+		subnetBInfo,
+		exampleMessengerB,
+		subnetAInfo,
+		exampleMessengerContractA,
+		exampleMessengerA,
+		fundedKey,
+		false)
+
+	utils.AddProtocolVersionAndWaitForAcceptance(
```

Comment:
Can you add comments to each of these statements mapping them to the test steps described at the beginning of the test function?

Reply:
added